src/functions/New-PuppetDscModule.ps1

```diff
@@ -70,7 +70,7 @@ Function New-PuppetDscModule {
     } Else {}
 
     # make sure that we're operating on a absolute path to avoid confusion from symlinks and relative paths
-    $OutputDirectory = New-Item -Path $OutputDirectory -ItemType "directory" -Force
+    $OutputDirectory = New-Item -Path "$OutputDirectory/$PuppetModuleName" -ItemType "directory" -Force
 
     $PuppetModuleRootFolderDirectory = Join-Path -Path $OutputDirectory                 -ChildPath $PuppetModuleName
     $VendoredDscResourcesDirectory   = Join-Path -Path $OutputDirectory                 -ChildPath "$PuppetModuleName/lib/puppet_x/dsc_resources"
```

Comment:
I _think_ this is the line we actually want to change - the path to the vendored modules, instead of the path to the overall parent output directory.